### DISCUSSION TITLE
How to Perform Operations on Each Chunk Before Uploading with ChunkedUploady

### DISCUSSION THREAD

**Discussion by datta202** [rpldy/react-uploady#717] [Q&A]
Hello,

I am using the `ChunkedUploady` package to upload files in chunks directly to S3. I want to perform some operations on each chunk before starting the upload. Specifically, I want to calculate the checksum of each chunk and include that checksum in the header.

Please note that calculating the checksum will take some time for each chunk, so we need to account for this delay.

Could you please tell me how I can achieve this?

**Comment by datta202** [comment]
Hi @yoavniran, is useChunkStartListener supports asynchronous callback?

![image](https://github.com/user-attachments/assets/06570b11-8b96-4a74-a890-1e5e8eed6da3)

I am getting typescript error
```
(data: ChunkStartEventData) => Promise<{     
            url?: string | undefined;     
            sendOptions?: SendOptions | undefined; 
}>
```

is not assignable to parameter of type

```
(data: ChunkStartEventData) => StartEventResponse
```

Type

```
Promise<{     
      url?: string | undefined;    
      sendOptions?: SendOptions | undefined; }>
```

is not assignable to type StartEventResponse

**Reply by yoavniran** [reply]
hey @datta202 
I think this is a typing bug on Uploady side. 
Most, if not all, event handling in Uploady is async by nature and the hooks should represent that. 

I will create a fix for this soon

**Reply by yoavniran** [reply]
@datta202 
fix is now available in https://github.com/rpldy/react-uploady/releases/tag/v1.8.3

Please let me know if there are any issues

**Reply by datta202** [reply]
Thank you so much @yoavniran!!

I will let you if there are any issues
